### PR TITLE
violin from the single cell gene expression plot

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Added use case to open a violin from the single cell gene expression plot

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/routes/termdb.singlecellSamples.ts [hunk 1]
```diff
@@ -184,6 +183,8 @@ function validateDataNative(D: SingleCellDataNative, ds: any) {
 }
 
 function validateGeneExpressionNative(G: SingleCellGeneExpressionNative) {
+	G.singleCellGeneExpression2bins = {} // cache for binning gene expression values
```
Comment (xzhou82) [inline server/routes/termdb.singlecellSamples.ts:186]: the cache should be {sample:{gene:binObj}} please document such and fix the logic

Comment (airenzp) [reply server/routes/termdb.singlecellSamples.ts:186]: yeah I missed that thanks

**Review by xzhou82 — APPROVED** [review]
thanks for the nice idea!

File: server/routes/termdb.singlecellSamples.ts (hunk 1 above)
Comment (xzhou82) [inline server/routes/termdb.singlecellSamples.ts:186]: no need to repeat `singleCell` in the name. suggest to rename to `sample2gene2expressionBin`

File: server/shared/types/dataset.ts [hunk 2]
```diff
@@ -368,6 +368,7 @@ export type SingleCellGeneExpressionNative = {
 	folder: string
 	/** dynamically added getter */
 	get?: (q: any) => any
+	singleCellGeneExpression2bins: { [sample: string]: { [gene: string]: any } }
```
Comment (xzhou82) [inline server/shared/types/dataset.ts:371]: change this property to optional

**Pull request merged**